### DISCUSSION TITLE
Need tips and tricks to diagnose upgrade problem

### DISCUSSION THREAD

**Discussion by dan-vystrcil** [argoproj/argo-cd#9154] [Q&A] [answered]
I just upgraded argocd from v2.2.5 to v2.3.3. There were no errors during the deployment and all the pods are running. I can log in and access the server from the CLI just fine and I see all my projects. The problem is that I can no longer access the UI from the browser and it's throwing a PR_CONNECT_RESET_ERROR message. I have poked at it enough and I need some tips and tricks to figure out what is happening.

Adding a bit more. Using localhost proxy does work. So argo server is serving up pages. It really sounds like a loadbalancer issue, but why should that change I am not sure.

More info, if I use HTTP and port 443 and add the path /argocd then it works. Similar combinations using HTTPS do not work.

in our install.yaml we had from our 2.2.5 install --rootpath /argocd, but none of us can recall why we needed to do that.

**Comment by todaywasawesome** [comment]
@da6d6i7-bronga your instinct that there is an issue with the loadbalancer sounds correct. How are you exposing the the service?

**Reply by dan-vystrcil** [reply]
Just using a simple load balancer. Here is the service with the ip's covered up.

``` yaml
% kubectl -n argocd get service/argocd-server -o yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"app.kubernetes.io/component":"server","app.kubernetes.io/name":"argocd-server","app.kubernetes.io/part-of":"argocd"},"name":"argocd-server","namespace":"argocd"},"spec":{"ports":[{"name":"http","port":80,"protocol":"TCP","targetPort":8080},{"name":"https","port":443,"protocol":"TCP","targetPort":8080}],"selector":{"app.kubernetes.io/name":"argocd-server"}}}
    service.beta.kubernetes.io/azure-load-balancer-internal: "true"
  creationTimestamp: "2022-02-16T18:47:41Z"
  finalizers:
  - service.kubernetes.io/load-balancer-cleanup
  labels:
    app.kubernetes.io/component: server
    app.kubernetes.io/name: argocd-server
    app.kubernetes.io/part-of: argocd
  name: argocd-server
  namespace: argocd
  resourceVersion: "26824355"
  uid: aed97322-d5e9-46b6-a1b2-b4bf17488141
spec:
  clusterIP: <w.x.y.z>
  clusterIPs:
  - <w.x.y.z>
  externalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: http
    nodePort: 30726
    port: 80
    protocol: TCP
    targetPort: 8080
  - name: https
    nodePort: 30218
    port: 443
    protocol: TCP
    targetPort: 8080
  selector:
    app.kubernetes.io/name: argocd-server
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - ip: <w.x.y.z>

**Comment by dan-vystrcil** [comment, marked as answer]
Solved:
I removed the additional commands from the install.yaml, re-applied, and what do you know... it works.

**Reply by sunyifei83** [reply]
I encountered the same problem. Could you provide more specific instructions on how to remove the additional commands from the install.yaml file?

**Reply by dan-vystrcil** [reply]
If I recall this correctly we were using --rootpath /argocd and then discovered that we did not need it. This is not a command that is in the install.yaml by default.
